### PR TITLE
006 - listLoyaltyTierRewards, Reward Assignments methods

### PR DESCRIPTION
**Pull request by weronika-kurczyna**
---
'@voucherify/sdk': minor
---

Added support for new endpoints and adding missing types in Rewards API.

Added support for new endpoints: 
- `/loyalties/{campaignId}/rewards/{assignmentId}`, (getRewardAssignment)
- `/loyalties/{campaignId}/tiers/{tierId}/rewards`, (listLoyaltyTierRewards)
- `/rewards/{rewardId}/assignments/{assignmentId}` (example available in readme.md)

New exported types/interfaces in `Loyalties.ts`: 
Domain types:
- LoyaltyTierRewardItem,
- LoyaltyTierRewardItemBase, 
- LoyaltyTierRewardItemParameters, 
- LoyaltyTierRewardItemCampaignParameters, 
- LoyaltyTierRewardItemCoinParameters, 
- LoyaltyTierRewardItemMaterialParameters, 
- LoyaltyTierRewardItemCampaignDiscountCoupons, 
- LoyaltyTierRewardItemCampaignGiftVouchersAndLoyaltyProgram

0-level types:
- LoyaltiesGetRewardAssignmentResponseBody,
- LoyaltiesListLoyaltyTierRewardsResponseBody

New exported types/interfaces in `Rewards.ts`: 
Domain types:
- RewardAssignment (old `RewardsAssignmentObject`), 
- RewardsAssignmentCampaignOrMaterialReward,
- RewardsAssignmentCoinReward
- 
0-level types:
- RewardsListAssignmentsRequestQuery (old `RewardsListAssignmentsParams`),
- RewardsListAssignmentsResponseBody (old `RewardsListAssignmentsResponse`), 
- RewardsCreateAssignmentRequestBody (old `RewardsCreateAssignment`), 
- RewardsCreateAssignmentCoinRewardRequestBody, 
- RewardsCreateAssignmentCampaignOrMaterialRewardRequestBody,
- RewardsCreateAssignmentResponseBody (old `RewardsCreateAssignmentResponse`), 
- RewardsUpdateAssignmentRequestBody (old `RewardsUpdateAssignment`), 
- RewardsUpdateAssignmentResponseBody (old `RewardsUpdateAssignmentResponse`)
- RewardsGetAssignmentResponseBody (brand new)

### CODE REVIEW

File: packages/sdk/src/types/Rewards.ts [hunk 1]
```diff
@@ -200,3 +201,83 @@ export interface RewardAssignment {
...
+	}
+}
+
+export interface RewardAssignmentResponseData {
+	created_at?: string
+	updated_at?: string | null
+	object?: 'reward-assignment'
+}
+
+export type RewardsAssignment = Required<RewardAssignmentIdentity> &
```
[the remark this answers is not in the document]
Comment (p-zielinski) [reply packages/sdk/src/types/Rewards.ts:229]: we use single-s in domain types?

File: packages/sdk/src/types/Rewards.ts [hunk 2]
```diff
@@ -137,8 +140,6 @@ export interface RewardRedemptionParams {
 	id?: string
 }
 
-// domain types
```
Comment (p-zielinski) [inline packages/sdk/src/types/Rewards.ts:140]: why you have deleted this ? under this line types are new. Domain types, not legacy

**Review by p-zielinski — CHANGES_REQUESTED** [review]

File: packages/sdk/src/types/Rewards.ts (hunk 2 above)
Comment (p-zielinski) [reply packages/sdk/src/types/Rewards.ts:140]: why?

File: packages/sdk/src/types/Loyalties.ts [hunk 3]
```diff
@@ -1219,3 +1219,94 @@ export interface EarningRuleProportionalCustomEvent {
 		}
 	}
 }
```
Comment (p-zielinski) [inline packages/sdk/src/types/Loyalties.ts:1221]: i see above not legacy types...

File: packages/sdk/src/types/Rewards.ts [hunk 4]
```diff
@@ -200,3 +201,83 @@ export interface RewardAssignment {
...
+	related_object_id?: string
+	related_object_type?: 'campaign'
+	parameters?: {
+		loyalty: {
+			points: number
+		}
+	}
+}
+
+export interface RewardAssignmentResponseData {
```
[the remark this answers is not in the document]
Comment (p-zielinski) [reply packages/sdk/src/types/Rewards.ts:224]: yep

File: packages/sdk/src/types/Rewards.ts [hunk 5]
```diff
@@ -200,3 +199,94 @@ export interface RewardAssignment {
...
+
+export type RewardsGetAssignmentResponseBody = RewardsAssignment
+
+// Create assignment
+
+export type RewardsCreateAssignmentRequestBody =
+	| RewardsCreateAssignmentCoinRewardRequestBody
+	| RewardsCreateAssignmentCampaignOrMaterialRewardRequestBody
+
+export interface RewardsCreateAssignmentCampaignOrMaterialRewardRequestBody {
```
Comment (p-zielinski) [inline packages/sdk/src/types/Rewards.ts:265]: does material reward not accept `validation_rules`. ?

**Review by p-zielinski — CHANGES_REQUESTED** [review]

File: packages/sdk/src/types/Rewards.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (weronika-kurczyna) [reply packages/sdk/src/types/Rewards.ts]: yep, there was already a type called RewardAssignment and that's why I used another name
but it turns out that `RewardAssignment` can be replaced with a new type, so it is no longer needed (RewardsAssignment was renamed to RewardAssignment)

**Review by p-zielinski — APPROVED** [review]

**Comment by marcin-slezak** [comment]
released as a part of other RC

**Pull request closed without merging**